### PR TITLE
Automated creating and updating scala_x_x.bzl repository files

### PR DESCRIPTION
**Pull request by lm1nrt**
### Description
- Updated SHA's, dependencies, artifacts for all of the scala_x_x.bzl repository files with transitive dependencies compliance
- Added a script that simplifies creating and updating scala_x_x.bzl repository files.

It's not the ideal version, but I believe it can be improved by adding support for more Scala artifacts.

### Motivation
Updating and creating new versions for scala_x_x.bzl files is time-consuming, and it's a kind of work that is not performed regularly due to its specifics.

### CODE REVIEW

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:36]: Some versions are not compatible/doesn't exist in certain Scala version

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:60]: The version location in maven coordinates isn't strict, so there are cases that the version is under the x+1 index.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:82]: There are 5 patterns of labels, which are related to the structure of certain group and artifact due to custom naming.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:131]: After mapping to the dict, there are no trailing commas, so this is a workaround to adjust file to .bzl files / starlark convention and CI-checks failures.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:164]: If file doesn't exist, the script is copying the last file from directory (default sort) and updating it.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:171]: There are artifacts versions that cause CI-check failures after update and exceptions in logs.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:191]: Not modifying the runtime dependencies due to CI-checks failure.

File: scripts/create_repository.py
Comment (lm1nrt) [inline scripts/create_repository.py:96]: Creates local file to get the directDependencies for fetched artifacts

**Comment by lukaszwawrzyk** [comment]
@liucijus @simuons Could you take a look at this PR? It is a first step to improve maven dependency management in rules_scala. It will allow to bump patch versions of scala and add new minor versions faster. Eventually we would like to organize dependencies more, to keep naming consistent, have clear understanding which deps belong into which resolution tree, etc. and have more frequent updates of other dependencies too.

**Comment by WojciechMazur** [comment]
@liucijus @simuons Ping for taking a look at this PR. 

It adds Scala 3.5.0, while at the same time, we're in the middle of the 3.5.2 release and its artifacts are already available, public release by the end of this week. https://github.com/scala/scala3/issues/21781
The 3.6.0 would be shipped in 5-6 weeks from now. 
Alternatively you could consider merging #1604

As a Scala 3 compiler team, we'd like to help maintain the rules and keep it up to date. The release plan for Scala 3 involves a new stable release every 6-8 weeks. We'd like to include ensuring the new version of compiler would be supported by Bazel scala_rules as part of our release procedure in the similar way as we're dealing currently with other vital parts of Scala tooling (eg. Debug Adapter Protocol, Metals, Scala CLI, Scastie)

Having a tool to easily add new version would be really beneficial here, but we'd also require a bit involvement from the maintainers of Scala rules to ensure changes would be reviewed and merged within a finite time span. I belive that if needed both me and @lukaszwawrzyk would consider getting the maintainer role in this project. If needed we can arrange some discussions on how @virtuslab as organization, can help with development of this project.

**Review by simuons — CHANGES_REQUESTED** [review]
Thanks, @lm1nrt, that looks good and useful. 
Sorry I made a conflict by merging https://github.com/bazelbuild/rules_scala/pull/1604 first.
 Could you please rebase this PR and I'll merge it once it's green.

**Review by simuons — APPROVED** [review]

**Comment by lm1nrt** [comment]
@simuons Done :) I thought that these changes caused CI failure: `This could either mean you have to add the '@@rules_java' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.`, but AFAIK it seems to be a global problem while looking on other PRs.

**Comment by simuons** [comment]
@lm1nrt, I'm looking why build is failing. This is not related to your changes. Failing task was supposed to be optional.

**Comment by simuons** [comment]
Hi, @lm1nrt, I'm really sorry but I'll ask you to rebase once again. Master is green now, so your PR should be as well.

**Pull request merged**
